### PR TITLE
FIX: De-duplicate poll vote on user merge

### PR DESCRIPTION
**Pull request by nattsw**
Currently when merging users, polls may error out if the source and target users have both voted on the same poll before. 😢 

There is currently no constraint on the `poll_votes` table either to support this. Ideally a composite primary key can be used `(poll_id, user_id)`, but alas there is no support yet, which is probably why it wasn't created in the first place.

This fix ensures that merging is successful by only keeping the target poll votes if duplicates exist.

This fix also runs a migration on older poll votes where failed merges would have caused a single user to have voted twice on a single poll. e.g. this weird edge case

Screenshot 2023-06-14 at 7 12 06 PM

Related: https://github.com/discourse/discourse/pull/22105

### CODE REVIEW

File: plugins/poll/spec/fabricators/poll_fabricator.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+Fabricator(:poll, class_name: "Poll") do
```
Comment (CvX) [inline plugins/poll/spec/fabricators/poll_fabricator.rb:3]: Class names aren't required here are they?

File: plugins/poll/spec/integration/user_merger_spec.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  fab!(:target_user) { Fabricate(:user, username: "galahad", email: "galahad@knights.com") }
+  fab!(:source_user) { Fabricate(:user, username: "lancelot", email: "lancelot@knights.com") }
+  fab!(:poll) { Fabricate(:poll) }
+
+  def merge_users!(source = nil, target = nil)
+    source ||= source_user
+    target ||= target_user
+    UserMerger.new(source, target).merge!
+  end
+
```
Comment (CvX) [inline plugins/poll/spec/integration/user_merger_spec.rb:13]: Unused?

```suggestion
```

**Review by CvX — APPROVED** [review]

File: plugins/poll/db/migrate/20230614041219_delete_duplicate_poll_votes.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+class DeleteDuplicatePollVotes < ActiveRecord::Migration[7.0]
+  def up
+    execute "
+    DELETE FROM poll_votes
+    WHERE (poll_id, user_id, updated_at) NOT IN (
+      SELECT poll_id, user_id, MAX(updated_at)
+      FROM poll_votes
+      GROUP BY poll_id, user_id
+    )"
```
Comment (CvX) [inline plugins/poll/db/migrate/20230614041219_delete_duplicate_poll_votes.rb:11]:
```suggestion
    execute <<~SQL
      DELETE FROM poll_votes
      WHERE (poll_id, user_id, updated_at) NOT IN (
        SELECT poll_id, user_id, MAX(updated_at)
        FROM poll_votes
        GROUP BY poll_id, user_id
      )
    SQL
```

**Comment by nattsw** [comment]
Thank you for reviewing @CvX

**Pull request merged**